csharp/src/iceboxnet/ServiceManager.cs

```diff
@@ -169,20 +174,15 @@ public void AddObserver(IServiceObserverPrx? observer, Ice.Current current)
                         _logger.Trace("IceBox.ServiceObserver", $"Added service observer {observer}");
                     }
 
-                    foreach (ServiceInfo info in _services)
-                    {
-                        if (info.Status == ServiceStatus.Started)
-                        {
-                            activeServices.Add(info.Name);
-                        }
-                    }
+                    activeServices = _services.Where(info => info.Status == ServiceStatus.Started).Select(
+                        info => info.Name).ToArray();
                 }
-            }
 
-            if (activeServices.Count > 0)
-            {
-                observer!.ServicesStartedAsync(activeServices.ToArray()).ContinueWith((t) => ObserverCompleted(observer, t),
-                    TaskScheduler.Current);
+                if (activeServices.Length > 0)
+                {
+                    observer.ServicesStartedAsync(activeServices).ContinueWith(
```

Comment:
Should we mark `addObserver` "amd" and use await on this call?

Reply:
This will delay `AddObserver` response util `ServicesStarted` complete